hwcomposer/qeglfsbackingstore.cpp

```diff
@@ -145,7 +151,7 @@ void QEglFSBackingStore::flush(QWindow *window, const QRegion &region, const QPo
         QRect imageRect = m_image.rect();
 
         QRegion fixed;
-        foreach (const QRect &rect, m_dirty.rects()) {
+        for (const QRect &rect : m_dirty) {
```

Comment:
This fails to build with Qt 5.6. QRegion::begin() needed for using for was introduced in Qt 5.8 https://doc.qt.io/qt-6/qregion.html#begin. This means we need to make this change conditional.

Reply:
This looks to be the same change as what I did on #95, should I close my PR in favor of this one? Thanks for testing this though ^^

Reply:
> This fails to build with Qt 5.6 wit error "error: 'begin' was not declared in this scope". QRegion::begin() needed for using for was introduced in Qt 5.8 https://doc.qt.io/qt-6/qregion.html#begin. This means we need to make this change conditional but this can be changed to "for (const QRect &rect : m_dirty.rects())" for Qt < 5.8 afaik. Same for the change below.

I'm happy with either, it's your code in the end, so whatever works for you I'm happy to adjust to that.

Reply:
I would go with "for (const QRect &rect : m_dirty.rects())" for Qt < 5.8 and "for (const QRect &rect : m_dirty)" for Qt >= 5.8 so the difference in the code is easier to see, I tested that the former works with Qt 5.6.

Reply:
@Herrie82 Any plans to fix the build issue mentioned above?